### ISSUE TITLE
Add ability to specific which pieces a member plays

### ISSUE THREAD

**Issue by dan-brewster** [csasf/members#22]
Add new table (1member_set_piece` or thereabouts, probably tied to a `performance_set_piece`) to represent which pieces in a `performance_set`.

By default, a member plays in all pieces. If they are unselected from playing in a piece, a reason must be given -- (1) instrument does not play in piece  (2) rotated out for other players to play.

This is important so we can fit as many members of stage during a performance as possible, within the overall limit.

**Comment by wenley** [comment]
Improvement: A member is only associated to a `PerformanceSet` through a `PerformancePiece` and the join table of `PerformancePiecesMembers` (Rails should have automagic helpers to generate this). It fits the idea that "you're part of a performance if and only if you play at least one piece", and no need for a validation then.

(Note: This is totally separate from a person being part of an ensemble - that feels like a separate real-world association, and thus deserves a separate data model association).

Migration should be pretty straight forward once `PerformanceSet`s have pieces populated - just default to everyone playing all pieces on a set and destroy the old `MemberSet` association.

**Comment by dan-brewster** [comment]
Yeah, I think this makes sense. There actually isn't a mapping `member` --> `ensemble` right now, because people switch between ensembles between sets, and we've found that there's not been a reason to sort folks in that way. We also do want to discourage people from feeling like the groups are completely separate entities, since each share players and mission, and we'd like them to have something of a join sense of community.

**Comment by dan-brewster** [comment]
PerformancePiecesMembers can't be completely magic -- that's where we'll need to store section information. And we'll need an index on that field as well... somewhat annoying.

The lookup to generate a program-style members list would be `PerformanceSet` --> (through `performance_set_id`) --> `PerformancePiece` --> (through `performance_piece_id`) --> `PerformancePiecesMembers` --> (through `member_id`) --> `Members`

**Comment by dan-brewster** [comment]
Based on further discussions, we will likely need a `MemberEnsemble` table. It'll depend largely on some hairy issues around figuring out how to recruit members for groups in the most fair fashion.
